File: app/src/main/java/com/sjani/usnationalparkguide/Data/ParkEntity.java
package com.sjani.usnationalparkguide.Data;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;


@Entity(tableName = "parks")
public class ParkEntity {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "park_id")
    private String park_id;
    @ColumnInfo(name = "park_name")
    private String park_name;
    @ColumnInfo(name = "states")
    private String states;
    @ColumnInfo(name = "parkCode")
    private String parkCode;
    @ColumnInfo(name = "latLong")
    private String latLong;
    @ColumnInfo(name = "description")
    private String description;
    @ColumnInfo(name = "designation")
    private String designation;
    @ColumnInfo(name = "address")
    private String address;
    @ColumnInfo(name = "phone")
    private String phone;
    @ColumnInfo(name = "email")
    private String email;
    @ColumnInfo(name = "image")
    private String image;
    @ColumnInfo(name = "isFav")
    private boolean isFav;


    @Override
    public String toString() {
        return "ParkEntity{" +
                "park_id='" + park_id + '\'' +
                ", park_name='" + park_name + '\'' +
                ", states='" + states + '\'' +
                ", parkCode='" + parkCode + '\'' +
                ", latLong='" + latLong + '\'' +
                ", description='" + description + '\'' +
                ", designation='" + designation + '\'' +
                ", address='" + address + '\'' +
                ", phone='" + phone + '\'' +
                ", email='" + email + '\'' +
                ", image='" + image + '\'' +
                ", isFav=" + isFav +
                '}';
    }

    public String getPark_id() {
        return park_id;
    }

    public void setPark_id(String park_id) {
        this.park_id = park_id;
    }

    public String getPark_name() {
        return park_name;
    }

    public void setPark_name(String park_name) {
        this.park_name = park_name;
    }

    public String getStates() {
        return states;
    }

    public void setStates(String states) {
        this.states = states;
    }

    public String getParkCode() {
        return parkCode;
    }

    public void setParkCode(String parkCode) {
        this.parkCode = parkCode;
    }

    public String getLatLong() {
        return latLong;
    }

    public void setLatLong(String latLong) {
        this.latLong = latLong;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDesignation() {
        return designation;
    }

    public void setDesignation(String designation) {
        this.designation = designation;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public boolean isFav() {
        return isFav;
    }

    public void setFav(boolean fav) {
        isFav = fav;
    }
}

File: app/src/main/java/com/sjani/usnationalparkguide/data/entity/ParkEntity.kt
package com.sjani.usnationalparkguide.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "parks")
data class ParkEntity(
    @PrimaryKey
    @ColumnInfo(name = "park_id")
    val parkId: String,
    
    @ColumnInfo(name = "park_name")
    val parkName: String = "",
    
    @ColumnInfo(name = "states")
    val states: String = "",
    
    @ColumnInfo(name = "parkCode")
    val parkCode: String = "",
    
    @ColumnInfo(name = "latLong")
    val latLong: String = "",
    
    @ColumnInfo(name = "description")
    val description: String = "",
    
    @ColumnInfo(name = "designation")
    val designation: String = "",
    
    @ColumnInfo(name = "address")
    val address: String = "",
    
    @ColumnInfo(name = "phone")
    val phone: String = "",
    
    @ColumnInfo(name = "email")
    val email: String = "",
    
    @ColumnInfo(name = "image")
    val image: String = "",
    
    @ColumnInfo(name = "isFav")
    val isFav: Boolean = false
)

@Entity(tableName = "fav_parks")
data class FavParkEntity(
    @PrimaryKey
    @ColumnInfo(name = "park_id")
    val parkId: String,
    
    @ColumnInfo(name = "park_name")
    val parkName: String = "",
    
    @ColumnInfo(name = "states")
    val states: String = "",
    
    @ColumnInfo(name = "parkCode")
    val parkCode: String = "",
    
    @ColumnInfo(name = "latLong")
    val latLong: String = "",
    
    @ColumnInfo(name = "description")
    val description: String = "",
    
    @ColumnInfo(name = "designation")
    val designation: String = "",
    
    @ColumnInfo(name = "address")
    val address: String = "",
    
    @ColumnInfo(name = "phone")
    val phone: String = "",
    
    @ColumnInfo(name = "email")
    val email: String = "",
    
    @ColumnInfo(name = "image")
    val image: String = ""
) {
    companion object {
        fun fromParkEntity(park: ParkEntity): FavParkEntity {
            return FavParkEntity(
                parkId = park.parkId,
                parkName = park.parkName,
                states = park.states,
                parkCode = park.parkCode,
                latLong = park.latLong,
                description = park.description,
                designation = park.designation,
                address = park.address,
                phone = park.phone,
                email = park.email,
                image = park.image
            )
        }
    }
}

@Entity(tableName = "trails")
data class TrailEntity(
    @PrimaryKey
    @ColumnInfo(name = "trail_id")
    val trailId: String,
    
    @ColumnInfo(name = "trail_name")
    val trailName: String = "",
    
    @ColumnInfo(name = "summary")
    val summary: String = "",
    
    @ColumnInfo(name = "difficulty")
    val difficulty: String = "",
    
    @ColumnInfo(name = "image_small")
    val imageSmall: String = "",
    
    @ColumnInfo(name = "image_med")
    val imageMed: String = "",
    
    @ColumnInfo(name = "length")
    val length: String = "",
    
    @ColumnInfo(name = "ascent")
    val ascent: String = "",
    
    @ColumnInfo(name = "latitude")
    val latitude: String = "",
    
    @ColumnInfo(name = "longitude")
    val longitude: String = "",
    
    @ColumnInfo(name = "location")
    val location: String = "",
    
    @ColumnInfo(name = "condition")
    val condition: String = "",
    
    @ColumnInfo(name = "more_info")
    val moreInfo: String = ""
)

@Entity(tableName = "camps")
data class CampEntity(
    @PrimaryKey
    @ColumnInfo(name = "camp_id")
    val campId: String,
    
    @ColumnInfo(name = "camp_name")
    val campName: String = "",
    
    @ColumnInfo(name = "description")
    val description: String = "",
    
    @ColumnInfo(name = "parkCode")
    val parkCode: String = "",
    
    @ColumnInfo(name = "address")
    val address: String = "",
    
    @ColumnInfo(name = "latLong")
    val latLong: String = "",
    
    @ColumnInfo(name = "cellPhoneReception")
    val cellPhoneReception: String = "",
    
    @ColumnInfo(name = "showers")
    val showers: String = "",
    
    @ColumnInfo(name = "internetConnectivity")
    val internetConnectivity: String = "",
    
    @ColumnInfo(name = "toilets")
    val toilets: String = "",
    
    @ColumnInfo(name = "wheelchairAccess")
    val wheelchairAccess: String = "",
    
    @ColumnInfo(name = "reservationsUrl")
    val reservationsUrl: String = "",
    
    @ColumnInfo(name = "directionsUrl")
    val directionsUrl: String = ""
)

@Entity(tableName = "alerts")
data class AlertEntity(
    @PrimaryKey
    @ColumnInfo(name = "alert_id")
    val alertId: String,
    
    @ColumnInfo(name = "alert_name")
    val alertName: String = "",
    
    @ColumnInfo(name = "description")
    val description: String = "",
    
    @ColumnInfo(name = "category")
    val category: String = "",
    
    @ColumnInfo(name = "parkCode")
    val parkCode: String = ""
)
